File: 백준/Gold/1206. 사람의 수/사람의 수.py
# 사람의 수
import sys
input = sys.stdin.readline

N = int(input())
scores = []

for i in range(N):
    integer, floating = input().rstrip().split(".")
    scores.append(int(integer) * 1000 + int(floating))

for i in range(1, 1001):
    cnt = 0

    for j in range(N):
        total = scores[j] * i

        if total % 1000 != 0:
            total = (total // 1000+1) * 1000
        
        if total // i == scores[j]:
            cnt += 1

    if cnt == N:
        break
        
print(i)
            

File: 백준/Gold/1206. 사람의 수/사람의 수.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());
        int[] scores = new int[N];
        int result = 0;

        for (int i = 0; i < N; i++) {
            String[] tmp = br.readLine().split("\\.");
            scores[i] = Integer.parseInt(tmp[0]) * 1000 + Integer.parseInt(tmp[1]);
        }

        for (int i = 1; i < 1001; i++) {
            int cnt = 0;
            result = i;

            for (int j = 0; j < N; j++) {
                int total = scores[j] * i;

                if (total % 1000 != 0 ) {
                    total = (total / 1000 + 1) * 1000;
                }

                if (total / i == scores[j]) {
                    cnt ++;
                }
            }

            if (cnt == N) {
                break;
            }
        }

        System.out.println(result);
    }
}
